geode-core/src/main/java/org/apache/geode/cache/query/internal/index/AbstractIndex.java

```diff
@@ -2002,7 +2002,8 @@ void populateListForEquiJoin(List list, Object outerEntries, Object innerEntries
    */
   synchronized void setPdxStringFlag(Object key) {
     // For Null and Undefined keys do not set the isIndexedPdxKeysFlagSet flag
-    if (key == null || key == IndexManager.NULL || key == QueryService.UNDEFINED) {
+    if (isIndexedPdxKeysFlagSet || key == null || key == IndexManager.NULL
```

Comment:
They are slightly different.  The isIndexedPdxKeys is represents whether the index is storing pdx as keys. The isIndexedPdxKeysFlagSet, is a boolean that is used as a short circuit to only call the method once.  I guess it was a performance "enhancement" to not call the method over and over for every value and just call it only for the first call.